tools/code-owners-parser/CodeOwnersParser/CodeOwnersFile.cs

```diff
@@ -9,28 +8,23 @@ public static class CodeOwnersFile
     {
         public static List<CodeOwnerEntry> ParseFile(string filePathOrUrl)
         {
-            string content;
-            content = FileHelpers.GetFileContents(filePathOrUrl);
-
+            string content = FileHelpers.GetFileContents(filePathOrUrl);
             return ParseContent(content);
         }
 
         public static List<CodeOwnerEntry> ParseContent(string fileContent)
         {
             List<CodeOwnerEntry> entries = new List<CodeOwnerEntry>();
-            string line;
-
 
             // An entry ends when we get to a path (a real path or a commented dummy path)
-
             using (StringReader sr = new StringReader(fileContent))
             {
                 CodeOwnerEntry entry = new CodeOwnerEntry();
 
                 // we are going to read line by line until we find a line that is not a comment OR that is using the placeholder entry inside the comment.
                 // while we are trying to find the folder entry, we parse all comment lines to extract the labels from it.
                 // when we find the path or placeholder, we add the completed entry and create a new one.
-                while ((line = sr.ReadLine()) != null)
+                while (sr.ReadLine() is { } line)
```

Comment:
My newer C# syntax is rusty. What is this statement actually trying to say? What does `{ }` in that expression represent?

Reply:
It is a special "empty" case of a [property pattern](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#property-pattern).

For example:

`static bool IsConferenceDay(DateTime date) => date is { Year: 2020, Month: 5, Day: 19 or 20 or 21 };`

but if you would do instead

`date is {};`

you would say "date is an object with _whatever_ properties." which effectively means "date is an object, so it is not null".

The `line` after `{}` is the name of the matched object, similar to this snippet, naming `s`:

```
static string TakeFive(object input) => input switch
{
    string { Length: >= 5 } s => s.Substring(0, 5),
```